internal/cloudapi/v2/openapi.v2.yml

```diff
@@ -211,6 +211,101 @@ paths:
               schema:
                 type: string
 
+  /composes/{id}/image:
```

Comment:
I've realized that I'm not sure if I like that we are calling this new entity an "image". My reasoning is that if you are a new person wanting to use this API, you will look for the "image" keyword which leads you to this new route. But often, creating a compose is just fine (e.g. for the koji usecase), you don't even need to copy the image in order to use it. I don't have a better name though... :thinking:

Reply:
Yea I agree the naming might be a bit confusing. First we had 'promotions', but the idea is that this is creating another image with the same content, just in a new location. And the only way to really guarantee that it's the same content is to copy/move around the blob (rebuilding an old manifest can have too many side effects, especially with user defined repos). 

The key is that you need a compose before you can create an image from it, and in the CRC api I think we should work towards an /images endpoint which parses both the compose result, and any images coming from that compose.

Reply:
Another option would be to just have a POST request  on /composes/{id}; and expand the GET request to include any copied and/or reshared images.

Reply:
What about calling it simply `copy`?

I agree that accepting `/composes/{id}` to accept POST and extending the GET seems a bit obscure.

Reply:
Well for s3 images this might become the endpoint to regenerate the s3 url on an object. Because of the image storage limitations of gcp, we might need to store those as a blob and this would become the endpoint to upload/register the image into gcp etc... So it's a bit more than just copy

Reply:
`/composes/{id}/share` might be more correct ?

Reply:
So, given your comments, the purpose of the endpoint is that exactly the same image build may get copied to a different region, shared with different accounts and there may be more going on in the background. But that should not be something to bother the user with (at least IMHO as it is an implementation detail if we e.g. reimport a blob from S3 to GCP). But the core point seems to be that the image won't be rebuilt, but used as it was built in the past. 

Funnily enough, the `/composes/{id}/share` endpoint seems restrictive to me, because more than just "sharing` could be happening - e.g. the image will be available in a new region.

Given all of that, I'll suggest what I didn't last time.  That is `/composes/{id}/clone`. This could be expressing that the original result of the compose is used as is and the ability to share it with different accounts or to copy to a different region does not seem too weird.

Reply:
I guess we'll end up with `/clones/{id}` then for a status endpoint?

@teg wdyt?

Reply:
Didn't see the last comment before I finished my review from yesterday, but I like it!

Reply:
Also @ondrejbudai wdyt?

Reply:
Works for me. :)